libs/x-content/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -23,21 +27,15 @@
 public class ParseField {
     private final String name;
     private final String[] deprecatedNames;
+    private final Set<RestApiCompatibleVersion> restApiCompatibleVersions = new HashSet<>();
```

Comment:
nit: initialize the size of HashSet to `2` (just a minor optimization since there will be at most 2 items)